src/main/java/picard/fingerprint/HaplotypeProbabilitiesFromGenotype.java

```diff
@@ -24,13 +24,26 @@
 
 package picard.fingerprint;
 
+import java.util.Arrays;
+
 /**
  * Represents a set of HaplotypeProbabilities that were derived from a single SNP
  * genotype at a point in time.
  */
 public class HaplotypeProbabilitiesFromGenotype extends HaplotypeProbabilities {
     private final Snp snp;
-    private final double[] likelihoods;
+    private double[] likelihoods;
+
+    @Override
+    public HaplotypeProbabilities deepCopy()  {
+        return new HaplotypeProbabilitiesFromGenotype(this);
+    }
+
+    public HaplotypeProbabilitiesFromGenotype(final HaplotypeProbabilitiesFromGenotype other){
+        super(other.getHaplotype());
```

Comment:
why not
```
why not
```
        this(snp, getHaplotype(), likelihoods[0], likelihoods[1], likelihoods[2]);
```